chkRanges.c

```diff
@@ -0,0 +1,74 @@
+#include "chkRanges.h"
+
+int ranges = 0;
+int chrgSession[] = {3, 3, 5, 4, 10, 11, 12};
+int rangeCount[3] = {1, 1, 1};
+
+void checkrange() {
+    int sizeOfchrgSess = sizeof(chrgSession) / sizeof(chrgSession[0]);
+    int sizeOfRngCnt = sizeof(rangeCount) / sizeof(rangeCount[0]);
+    sortRange(chrgSession, 0, sizeOfchrgSess-1);
+    countRange(chrgSession, sizeOfchrgSess);
+    printRange(rangeCount, sizeOfRngCnt-1);
+}
+
+void swapValues(int* value1, int* value2) 
+{ 
+    int temp = *value1; 
+    *value1 = *value2;
+    *value2 = temp; 
+}   
+
+int partition(int array[], int lowIndex, int highIndex) 
+{ 
+    int pivotElement = array[highIndex];
+    int i = (lowIndex - 1);
+    for (int index = lowIndex; index <= highIndex- 1; index++) {
+        if (array[index] <= pivotElement) {
+            i++;
+            swapValues(&array[i], &array[index]);
+        }
+    }
+    swapValues(&array[i + 1], &array[highIndex]);
+    return (i + 1);
+} 
+
+void sortRange(int array[], int lowIndex, int highIndex)
+{  
+    if (lowIndex < highIndex) {
+        int pivot = partition(array, lowIndex, highIndex);
+        sortRange(array, lowIndex, pivot - 1);
+        sortRange(array, pivot + 1, highIndex);
+    }
+} 
+
+bool compareRange(int CurPosVal, int IncPosVal, int CurPosIncVal)
+{
+    bool limit1 = (IncPosVal == CurPosVal);
+    bool limit2 = (IncPosVal == CurPosIncVal);
+    return (limit1 || limit2);
+}
+
+void countRange(int array[], int size)
```

Comment:
Better to pass the `rangeCount` also as a parameter to the function. Having it global means the program cannot execute multi-threaded. Also, one call will influence the output of the previous one

Reply:
Yes, I missed multi threading view. Corrected now.